packages/dds/merge-tree/src/test/testClientLogger.ts

```diff
@@ -197,7 +197,8 @@ export class TestClientLogger {
                         if (toRemovalInfo(seg) === undefined) {
                             const segProps = seg.properties;
                             for (let i = 0; i < seg.cachedLength; i++) {
-                                if (!matchProperties(segProps, properties[pos + i])) {
```

Comment:
rather than a complex condition inline can we just make a function in this file that does match properties, but also takes into account empty === undefined.